### PR TITLE
add the user id support of runAsUser

### PR DESCRIPTION
**Pull request by zhangzhenhao**
set uid if runAsUser has been seted.

### CODE REVIEW

File: pkg/server/container_create.go [hunk 1]
```diff
@@ -222,6 +222,12 @@ func (c *criContainerdService) generateContainerSpec(id string, sandboxPid uint3
 	// TODO(random-liu): [P1] Set selinux options.
 
 	// TODO(random-liu): [P1] Set user/username.
+	// NOTE(zzh): only set the user id.
+	runAsUser := securityContext.GetRunAsUser()
+	if runAsUser != nil {
+		g.SetProcessUID(uint32(runAsUser.GetValue()))
+
```
Comment (Random-Liu) [inline pkg/server/container_create.go]: unnecessary empty line.

File: pkg/server/container_create.go [hunk 2]
```diff
@@ -222,6 +222,12 @@ func (c *criContainerdService) generateContainerSpec(id string, sandboxPid uint3
 	// TODO(random-liu): [P1] Set selinux options.
 
 	// TODO(random-liu): [P1] Set user/username.
+	// NOTE(zzh): only set the user id.
```
Comment (Random-Liu) [inline pkg/server/container_create.go]: Just pure comment is ok here. No need to add `NOTE`.

You can add TODO for the user name, and remove the previous TODO.

**Comment by Random-Liu** [comment]
@zhangzhenhao as @mikebrow said, please remove corresponding SKIP from https://github.com/kubernetes-incubator/cri-containerd/blob/master/hack/test-cri.sh#L25.

Note that if I remember correctly, there are 2 `RunAsUser` test. One for uid, one for username. You may want to still skip username test, but enable uid one.

File: pkg/server/container_create.go (hunk 2 above)
Comment (zhangzhenhao) [inline pkg/server/container_create.go]: Hi Random-Liu,  Thanks for your review, I will do as your suggestion.

File: pkg/server/container_create.go (hunk 1 above)
Comment (zhangzhenhao) [inline pkg/server/container_create.go]: Hi Random-Liu,  Thanks for your review,  I will remove it

**Comment by zhangzhenhao** [comment]
hi @mikebrow , thanks a lot for your review. I will do this follow your suggestion. 
@Random-Liu thanks for your kindly reminder.

**Comment by Random-Liu** [comment]
@zhangzhenhao FYI, https://github.com/containerd/containerd/issues/1360#issuecomment-323849342.

**Comment by Random-Liu** [comment]
@zhangzhenhao Hm, seems like something wrong.

**Comment by zhangzhenhao** [comment]
hi @Random-Liu，I am so sorry for that，chould you  give me more detail about the issue. I will try my best to fix it ASAP.  Thanks for your help.

**Comment by mikebrow** [comment]
You last commit went awry. Probably needed to do the following before pushing your commit:
git checkout master
git fetch upstream master --tags
git merge upstream/master
git push origin master
That resets your master branch in your origin..
Then do this:
git checkout yourBranch
git rebase master
if there are conflicts you'll need to fix them 
then git add the modified files.. then
git rebase --continue 
or something like that .. git will give you instructions..
Then finally:
git push origin yourBranch

if you have Questions pls ping me on slack mikebrow on Kubernetes

**Comment by Random-Liu** [comment]
@mikebrow Thanks for the detailed instruction. You are so kind. :D

**Comment by Random-Liu** [comment]
LGTM, please squash and sign your commits.

`git rebase -i HEAD~7` and edit to merge 7 commits into one.
`git commit --amend -s` to sign your commit.

**Comment by zhangzhenhao** [comment]
@mikebrow  Thanks a lot for your kinder detailed instruction. And I am so sorry for my fault. Thanks again for your review and help.

**Comment by Random-Liu** [comment]
@zhangzhenhao You have too many commits here. You need to squash them.

**Comment by zhangzhenhao** [comment]
@Random-Liu ,I'm so sorry to trouble you again and again. I has double check my update and test it again for make sure it can work.  And, Push a new commit , follow @mikebrow 's suggestion.   Thanks again.

**Comment by Random-Liu** [comment]
@zhangzhenhao Congs! :)

**Pull request merged**